db/migrate/20171004112117_add_pam_name_to_users.rb

```diff
@@ -0,0 +1,5 @@
+class AddPamNameToUsers < ActiveRecord::Migration[5.1]
+  def change
+    add_column :users, :pam_name, :string, null: true, default: nil
```

Comment:
You have a unique validation on this column, but it's not safe against race conditions. I suggest a unique index as a db-level constraint.